### PR TITLE
docs(GKO): update Kubernetes operator context doc

### PR DESCRIPTION
**Pull request by Okhelifi**
**Issue**

- https://gravitee.atlassian.net/browse/APIM-290
- https://gravitee.atlassian.net/browse/APIM-600
- https://gravitee.atlassian.net/browse/APIM-543

**Description**

This PR includes all doc updates necessary following in this issue [APIM-290](https://gravitee.atlassian.net/browse/APIM-290).

The main purpose here is to remove all direct reference to the CRD ManagementContext that as been updated to become the ApiContext. We also introduce the templating using key value map in the ApiContext crd.

⚠️ Some links here reference files that are not present on the master branch. I assume that everything should be ok when we will do our next release. For the moment everything is present on the alpha branch. We can way for the release before merging this doc.

**Additional context**

[APIM-290]: https://gravitee.atlassian.net/browse/APIM-290?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://graviteedocs.blob.core.windows.net/docs-update-gko-context-doc/index.html)

### CODE REVIEW

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    organizationId: acme
+    auth:
+      auth:
+        username: admin
+        password: admin
+  values:
+    env: dev
+----
+
+{% raw %}
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:150]: What does this line do? It doesn't seem to have any impact on the HTML output.

Comment (Okhelifi) [reply pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:150]: it avoids all templatings like ``{{.Values.env}}`` in the following yaml to be interpreted

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    environmentId: dev
+    organizationId: acme
+    auth:
+      secretRef:
+        name: management-context-credentials
+----
+
+NOTE: If no namespace has been specified for the secret reference, the management context resource namespace will be used to resolve the secret.
+
+NOTE: To target another environment on the same API instance, add another Management Context configured to do that.
```
Comment (a-cordier) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:62]: API Context ?

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+  management:
+    baseUrl: https://gravitee-api.acme.com
+    environmentId: dev
+    organizationId: acme
+    auth:
+      secretRef:
+        name: management-context-credentials
+----
+
+NOTE: If no namespace has been specified for the secret reference, the management context resource namespace will be used to resolve the secret.
```
Comment (a-cordier) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:60]: API context

**Comment by Okhelifi** [comment]
> Which links reference files that aren't on the `master` branch? If you let me know I will investigate further.

For example this one https://raw.githubusercontent.com/gravitee-io/gravitee-kubernetes-operator/master/config/samples/context/dev/api-context-with-credentials.yml does not exist. But it's normal. There is nothing to investigate for the moment. 
The file exist in the alpha branch. When we will merge the alpha branch in master everything will be ok

**Comment by AndrewMcFarlandCampbellGravitee** [comment]
> > Which links reference files that aren't on the `master` branch? If you let me know I will investigate further.
> 
> For example this one [gravitee-io/gravitee-kubernetes-operator@`master`/config/samples/context/dev/api-context-with-credentials.yml (raw)](https://raw.githubusercontent.com/gravitee-io/gravitee-kubernetes-operator/master/config/samples/context/dev/api-context-with-credentials.yml) does not exist. But it's normal. There is nothing to investigate for the moment. The file exist in the alpha branch. When we will merge the alpha branch in master everything will be ok

OK. I was worried that you meant the `master` branch of `gravitee-docs`.

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc (hunk 1 above)
Comment (AndrewMcFarlandCampbellGravitee) [reply pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:150]: This section looks very odd when rendered as HTML. The text above mentions "The example below", but there seem to be two examples. Is that intentional? If it is, there needs to be more text explaining the examples.

Comment (Okhelifi) [reply pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:150]: In the operator the Api Context is created only to be used with an ApiDefinition. Here we explain how to includes key values in the ApiContext. That's why there is two yaml. The user will need both examples to be able to do some templating. But I can rephrase to make it more clear

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc (hunk 2 above)
Comment (Okhelifi) [reply pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:62]: Done

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc (hunk 3 above)
Comment (Okhelifi) [reply pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:60]: Done

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-installation-cluster.adoc [hunk 4]
```diff
@@ -12,15 +12,20 @@
...
-Before you start the deployment process, ensure that you have access to an APIM-ready Kubernetes cluster.
+=== Prerequisites
+
+Before you start, you need to have the following software set up and running on your machine:
+
+. https://kubernetes.io/docs/tasks/tools/#kubectl[Kubectl].
+. https://helm.sh/docs/intro/install/[Helm v3].
 
-By default the Kubernetes synchronizer is configured to look for API definitions in the API Gateway namespace. To watch all namespaces you must set the property `services.sync.kubernetes.namespaces` to `all` in the Gateway configuration. You can also provide a specific list of namespaces to watch.
+Moreover, ensure that you have access to an APIM-ready Kubernetes cluster.
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-installation-cluster.adoc:26]:
```suggestion
You also need access to an APIM-ready Kubernetes cluster.
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-quick-start.adoc [hunk 5]
```diff
@@ -110,9 +122,7 @@ NOTE: For the Gateway to work with the GKO, ensure that the `services.sync.kuber
 To test the API, you can call it through the APIM Gateway by running the following command using your APIM Gateway URL:
 
 ....
-curl -i http://localhost:9000/gateway/k8s-basic-with-ctx
+curl -i http://<YOUR_GATEWAY_URL>/gateway/k8s-basic-with-ctx
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-quick-start.adoc:125]:
```suggestion
curl -i http://<gateway_host>:<gateway_port>/gateway/k8s-basic-with-ctx
```
And below say "Where `<gateway_host>` and `<gateway_port>` are the gateway host and port.

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-installation-local.adoc [hunk 6]
```diff
@@ -1,105 +0,0 @@
-[[apim-kubernetes-operator-installation-local]]
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-installation-local.adoc:1]: Why has this page been deleted?

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-installation.adoc [hunk 7]
```diff
@@ -1,46 +0,0 @@
-[[apim-kubernetes-operator-installation]]
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-installation.adoc:1]: Why has this page been deleted?

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-installation-local.adoc (hunk 6 above)
Comment (Okhelifi) [reply pages/apim/3.x/kubernetes/apim-kubernetes-operator-installation-local.adoc:1]: Every references to the installation in a local cluster using dev tools like k3d should not be referenced in the official documentation. It makes the usage of our product and the documentation confusing. We agreed in the team that everything related to dev tools should be only documented on GitHub in the project documentation.

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-installation-cluster.adoc [hunk 8]
```diff
@@ -37,26 +42,31 @@ NOTE: See the link:https://kubernetes.io/docs/reference/kubectl/cheatsheet/#kube
...
-clusterrole.rbac.authorization.k8s.io/gko-manager-role created
-clusterrole.rbac.authorization.k8s.io/gko-metrics-reader created
-clusterrole.rbac.authorization.k8s.io/gko-proxy-role created
-rolebinding.rbac.authorization.k8s.io/gko-leader-election-rolebinding created
-clusterrolebinding.rbac.authorization.k8s.io/gko-manager-rolebinding created
-clusterrolebinding.rbac.authorization.k8s.io/gko-proxy-rolebinding created
-configmap/gko-manager-config created
-service/gko-controller-manager-metrics-service created
-deployment.apps/gko-controller-manager created
+Thank you for installing The Gravitee Kubernetes Operator Helm Chart !
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-installation-cluster.adoc:60]:
```suggestion
Thank you for installing The Gravitee Kubernetes Operator Helm Chart!
```

**Review by AndrewMcFarlandCampbellGravitee — APPROVED** [review]

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+:page-permalink: apim/3.x/apim_kubernetes_operator_user_guide_api_context.html
+:page-folder: apim/kubernetes
+:page-layout: apim3x
+
+[label label-version]#New in version 3.19.0#
+[label label-version]#BETA release#
+
+== How to use the Api Context (`ApiContext`) custom resource
+
+To enable synchronization of CRDs with a remote link:https://docs.gravitee.io/apim/3.x/apim_overview_architecture.html[management API^], you need to create an Api Context referring to an existing link:https://docs.gravitee.io/am/current/am_adminguide_organizations_and_environments.html[organization and environment^].
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:13]:
```suggestion
To enable synchronization of CRDs with a remote link:/apim/3.x/apim_overview_architecture.html[management API^], you need to create an Api Context referring to an existing link:/am/current/am_adminguide_organizations_and_environments.html[organization and environment^].
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+The management object can authenticate to your management API instance by using either basic authentication or a bearer token. Authentication credentials may either be added inline in the Management Context definition or referenced from a Kubernetes secret.
+
+NOTE: If both credentials and a bearer token are defined in your custom resource, the bearer token will take precedence.
+
+=== The Values object
+The values object is used to define a key value map. Those values are used to make your Kubernetes manifest reusable and dynamic. You can use link:https://pkg.go.dev/text/template?utm_source=godoc[Go templating] to build your Api Definition manifests.
+
+== Examples
+
+The custom resource in the example below refers to a Management API instance exposed at `+https://gravitee-api.acme.com+`, and targets the `dev` environment of the `acme` organization, with the `admin` account, using basic authentication credentials defined in a Kubernetes secret:
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:29]:
```suggestion
The custom resource in the example below refers to a Management API instance exposed at `+https://gravitee-api.example.com+`, and targets the `dev` environment of the `example-org` organization, with the `admin` account, using basic authentication credentials defined in a Kubernetes secret:
```
`example.com` should be used for examples in documentation. See  [RFC2606](https://www.rfc-editor.org/rfc/rfc2606.html). `acme.com` should be replaced with `example.com` throughout, and `acme` should be replaced with `example-org` throughout.

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+[,yaml]
+----
+apiVersion: gravitee.io/v1alpha1
+kind: ApiContext
+metadata:
+  name: apim-example-context
+  namespace: graviteeio
+spec:
+  management:
+    baseUrl: https://gravitee-api.acme.com
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:52]:
```suggestion
    baseUrl: https://gravitee-api.example.com
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+    environmentId: dev
+    organizationId: acme
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:54]:
```suggestion
    organizationId: example-org
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc [hunk 13]
```diff
@@ -0,0 +1,202 @@
...
+[,yaml]
+----
+apiVersion: gravitee.io/v1alpha1
+kind: ApiContext
+metadata:
+  name: apim-example-context
+  namespace: graviteeio
+spec:
+  management:
+    baseUrl: https://gravitee-api.acme.com
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:75]:
```suggestion
    baseUrl: https://gravitee-api.example.com
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc [hunk 14]
```diff
@@ -0,0 +1,202 @@
...
+    environmentId: dev
+    organizationId: acme
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:77]:
```suggestion
    organizationId: example-org
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc [hunk 15]
```diff
@@ -0,0 +1,202 @@
...
+
+[,yaml]
+----
+apiVersion: gravitee.io/v1alpha1
+kind: ApiContext
+metadata:
+  name: apim-example-context
+spec:
+  management:
+    baseUrl: https://gravitee-api.acme.com
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:120]:
```suggestion
    baseUrl: https://gravitee-api.example.com
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc [hunk 16]
```diff
@@ -0,0 +1,202 @@
...
+    environmentId: staging
+    organizationId: acme
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:122]:
```suggestion
    organizationId: example-org
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc [hunk 17]
```diff
@@ -0,0 +1,202 @@
...
+[,yaml]
+----
+apiVersion: gravitee.io/v1alpha1
+kind: ApiContext
+metadata:
+  name: context-dev
+  namespace: graviteeio
+spec:
+  management:
+    baseUrl: https://gravitee-api.acme.com
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:139]:
```suggestion
    baseUrl: https://gravitee-api.example.com
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc [hunk 18]
```diff
@@ -0,0 +1,202 @@
...
+    environmentId: dev
+    organizationId: acme
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:141]:
```suggestion
    organizationId: example-org
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc [hunk 19]
```diff
@@ -0,0 +1,202 @@
...
+[,yaml]
+----
+apiVersion: gravitee.io/v1alpha1
+kind: ApiContext
+metadata:
+  name: context-staging
+  namespace: graviteeio
+spec:
+  management:
+    baseUrl: https://gravitee-api.acme.com
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:160]:
```suggestion
    baseUrl: https://gravitee-api.example.com
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc [hunk 20]
```diff
@@ -0,0 +1,202 @@
...
+    environmentId: staging
+    organizationId: acme
```
Comment (AndrewMcFarlandCampbellGravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:162]:
```suggestion
    organizationId: example-org
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc (hunk 10 above)
Comment (Okhelifi) [reply pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc:29]: All acme has been replaced 👍🏼

**Review by AndrewMcFarlandCampbellGravitee — APPROVED** [review]

**Pull request closed without merging**